### PR TITLE
[WIP] Fix unit test to match comments - note this is gauranteed to fail :)

### PR DESCRIPTION
**Pull request by jayunit100**
First part of the fix for #33198 . 

Gauranteed to fail - as the unit tst is incorrect in its design.  

This just fixes the test so that it labels china in the 2nd machine.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33199)

### CODE REVIEW

**Comment by jayunit100** [comment]
Well, im new to this but... 

from `processTerms`  in the interpod_affinity.go module: 
```
for _, node := range p.nodes {
				if p.failureDomains.NodesHaveSameTopologyKey(node, fixedNode, term.TopologyKey) {
					p.counts[node.Name] += weight
				}
			}
```

Looks suspicious.  It seems like we need more than one if statement if we want to score pod affinity based on *other* pods that are running in the same node.  I guess we should think that maybe, `processTerms` incomplete at this time?  

@wojtek-t will know i guess.

File: plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go [hunk 1]
```diff
@@ -97,6 +97,7 @@ func (p *podAffinityPriorityMap) processTerm(term *api.PodAffinityTerm, podDefin
 				if p.failureDomains.NodesHaveSameTopologyKey(node, fixedNode, term.TopologyKey) {
 					p.counts[node.Name] += weight
 				}
+				// TODO i think we need to  add processing logic for pod labels here?
```
Comment (wojtek-t) [inline plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go]: I don't understand this TODO. It's obviously possible that there is something wrong here, but pod labels are being processed in "PodMatchesTermsNamespaceAndSelector" function ~10 lines above.

Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go]: Ok I will look at the code, at first glance though, I think ProcessTerm should be, by definition, involvving the pod matches term namespace / selector information contents in its calculation.

Comment (wojtek-t) [inline plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go]: >
I think ProcessTerm should be, by definition, involvving the pod matches term namespace / selector information contents in its calculation.

Yes - and it involes. This is delegated PodMatchesTermsNamespaceAndSelector which is called ~10 lines above __as part of processTerm__ function.

Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go]: could be the  culprit, in the code now will see whats going on

Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go]: ahhh i see whats going on.  the region stuff isnt in the labels, its separate i think.  /me digs more

Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go]: You know what?  I think this is really handled at the node level.

The bug here is that the test is expecting this to be handled under *interpod_affinity* priorities, but really, its handled as part of the *node_affinity* logic.

Does that sound about right?  If so i think its a one line fix :)

File: plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go [hunk 2]
```diff
@@ -102,10 +103,24 @@ func (p *podAffinityPriorityMap) processTerm(term *api.PodAffinityTerm, podDefin
...
 
-func (p *podAffinityPriorityMap) processTerms(terms []api.WeightedPodAffinityTerm, podDefiningAffinityTerm, podToCheck *api.Pod, fixedNode *api.Node, multiplier int) {
+func (p *podAffinityPriorityMap) processTerms(terms[] api.WeightedPodAffinityTerm, podDefiningAffinityTerm, podToCheck *api.Pod, fixedNode *api.Node, multiplier int, podlister algorithm.PodLister) {
 	for i := range terms {
 		term := &terms[i]
-		p.processTerm(&term.PodAffinityTerm, podDefiningAffinityTerm, podToCheck, fixedNode, float64(term.Weight*multiplier))
+
+
+		// A crude implementation of on-node pod label selector scoring.
+		selector,_ := unversioned.LabelSelectorAsSelector(term.PodAffinityTerm.LabelSelector)
```
Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go]: @wojtek-t As a first hack/pass  I tried scanning all the pods which match the selector - and increasing node scores if one matching pod is found. This  Seems to mostly effect the behaviour we want.   It adds more subtelety to the scoring algorithm though ... not sure the right way to weight .

File: plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go [hunk 3]
```diff
@@ -102,10 +103,24 @@ func (p *podAffinityPriorityMap) processTerm(term *api.PodAffinityTerm, podDefin
...
+		pods ,_ := podlister.List(selector)
+		for _, mypod := range pods {
+			if mypod.Spec.NodeName == fixedNode.Name {
+				p.Lock()
+				p.counts[fixedNode.Name] += float64(term.Weight * multiplier)
```
Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go]: i think this should be a non multiplied, but iirc there was a reason for the multiplication here , i.e. just to spread the scores

**Comment by jayunit100** [comment]
Closing because obsolete, but hope we can circle back to this semantic bug sometime

**Pull request closed without merging**